### PR TITLE
docs(concepts): rendering

### PR DESCRIPTION
**Pull request by clemcode**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR introduces an explanation of different rendering modes supported by Nuxt in the Concepts section

### CODE REVIEW

**Review by danielroe — APPROVED** [review]
Love it! ❤️

File: docs/content/2.concepts/2.rendering.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Rendering modes
+
+Browsers and servers can interpret Javascript code to turn Vue components into actual HTML elements. This step is called **rendering**. Nuxt supports both **client-side** and **universal** rendering, the two approaches have pros and cons that we will cover in this section.
+
+## Client-side Rendering
+
+Out of the box, a traditional Vue application is rendered in the browser (or **client**). The HTML is built by Vue after the browser downloads and parses all the Javascript code containing the instructions to build the current interface.
+
+![Users have to wait for the javascript to be downloaded and parsed before seeing the page's content](/img/concepts/rendering/light/csr.svg){.dark:hidden}
+![Users have to wait for the javascript to be downloaded and parsed before seeing the page's content](/img/concepts/rendering/dark/csr.svg){.light:hidden}
```
Comment (pi0) [inline docs/content/2.concepts/2.rendering.md:10]:
```suggestion
![Users have to wait for the javascript to be downloaded, parsed and executated before seeing the page's content](/img/concepts/rendering/dark/csr.svg){.light:hidden}
```

File: docs/content/2.concepts/2.rendering.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Rendering modes
+
+Browsers and servers can interpret Javascript code to turn Vue components into actual HTML elements. This step is called **rendering**. Nuxt supports both **client-side** and **universal** rendering, the two approaches have pros and cons that we will cover in this section.
+
+## Client-side Rendering
+
+Out of the box, a traditional Vue application is rendered in the browser (or **client**). The HTML is built by Vue after the browser downloads and parses all the Javascript code containing the instructions to build the current interface.
+
+![Users have to wait for the javascript to be downloaded and parsed before seeing the page's content](/img/concepts/rendering/light/csr.svg){.dark:hidden}
```
Comment (pi0) [inline docs/content/2.concepts/2.rendering.md:9]:
```suggestion
![Users have to wait for the javascript to be downloaded and parsed, executated before seeing the page's content](/img/concepts/rendering/light/csr.svg){.dark:hidden}
```

File: docs/content/2.concepts/2.rendering.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+![Users have to wait for the javascript to be downloaded and parsed before seeing the page's content](/img/concepts/rendering/dark/csr.svg){.light:hidden}
+
+While this technique allows to build complex and dynamic UIs with smooth page transitions, it has different pros and cons:
+
+### Pros
+
+- **Development speed**: When working fully on the client, we don’t have to worry about the server compatibility of the code, for example by using browser-only APIs like the `window` object.
+- **Cheaper:** Running a server adds a cost of infrastructure as you would need to run on a platform that supports Javascript. Client-only applications can be hosted on any static server, just as HTML or CSS files.
```
Comment (pi0) [inline docs/content/2.concepts/2.rendering.md:17]: I would also list offline usage in the "Pros" section

File: docs/content/2.concepts/2.rendering.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+### Cons
+
+- **Performance** : The user has to wait for the browser to download and parse javascript files. Depending on the network for the download part and the user's device for the parsing and execution, this can take some time and impact your users experience.
+- **Search Engine Optimization** : Indexing the content delivered via client-side rendering takes more time than an HTML document already built. This is related to the performance drawback we talked about, as search engine crawler won't wait for the interface to be fully displayed in their first try to index the page. With pure client-side rendering, your content will take more time to show up in Search results pages.
```
Comment (pi0) [inline docs/content/2.concepts/2.rendering.md:22]:
```suggestion
- **Search Engine Optimization** : Indexing and updating the content delivered via client-side rendering takes more time than an HTML document already built. This is related to the performance drawback we talked about, as search engine crawler won't wait for the interface to be fully displayed in their first try to index the page. With pure client-side rendering, your content will take more time to show up in Search results pages.
```

File: docs/content/2.concepts/2.rendering.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+### Examples
+
+Client-side rendering is a good choice for heavily interactive **web applications** that don’t need indexing, or will be frequently used by the same users as it can leverage browser caching to skip the download phase on subsequent visits, such as **SaaS, back office applications or Web Games**.
+
+## Universal Rendering
+
+With universal rendering, when the browser requests it, a fully rendered HTML page is returned to the browser. Whether the page has been generated in advance and cached or whether it is rendered on the fly, at some point Nuxt has run the Javascript (Vue) code in a server environment, producing an HTML document. Users immediately get the content of our application, contrary to client-side rendering. This step is similar to traditional **server-side rendering** performed by PHP or Ruby applications.
```
Comment (pi0) [inline docs/content/2.concepts/2.rendering.md:30]:
```suggestion
With universal rendering, when the browser requests a URL, a fully rendered HTML page is returned to the browser. Whether the page has been generated in advance and cached or whether it is rendered on the fly, at some point Nuxt has run the Javascript (Vue) code in a server environment, producing an HTML document. Users immediately get the content of our application, contrary to client-side rendering. This step is similar to traditional **server-side rendering** performed by PHP or Ruby applications.
```

File: docs/content/2.concepts/2.rendering.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+![Users can access the static content when the HTML document is loaded. Hydration then allows page's interactivity](/img/concepts/rendering/dark/ssr.svg){.light:hidden}
+
+### Pros
+
+- **Performance**: Users can get immediate access to the page’s content, because browsers can display static content much faster than Javascript-generated one. At the same time, Nuxt preserves the interactivity of a web application when the hydration process happens.
+- **SEO:** Universal rendering delivers the full HTML content of the page to the browser as a classic server application. Web crawlers can directly index the content of the page, which makes Universal rendering a great choice for any content that you want to index quickly.
+
+### Cons
+
+- **Development constraints:** Server and browser environments don’t provide the same APIs, and it can be tricky to write code that can run on both sides seamlessly. Fortunately, Nuxt provides guidelines and specific variables to help you determine where a piece of code is executed.
```
Comment (pi0) [inline docs/content/2.concepts/2.rendering.md:48]: I would also list additional cost of server rendering

File: docs/content/2.concepts/2.rendering.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+Hybrid rendering gives the ability to use different caching rules per routes, and decide how the server should respond to a new request on a given route.
+
+At the moment, every page (or **route**) of a Nuxt application must use the same rendering mode, client-side or universal. But in various cases, some pages could be generated at build time, while other should be client-side rendered. For example, think of a content website with an admin section. Every content page should be mostly static and generated once, but the admin section requires registration and behaves more like a dynamic application.
+
+[Read the open RFC discussing implementation and gathering community feedback](https://github.com/nuxt/framework/discussions/560)
+
+### Edge-side Rendering
+
+Nitro is the new [server engine](https://v3.nuxtjs.org/concepts/server-engine) that powers Nuxt 3. It offers cross-platform support for Node.js, Deno, Workers and more. Nitro has been designed with serverless rendering in mind, and allows to render a Nuxt application at the edge: closer to your users, allowing replication and further optimizations.
```
Comment (pi0) [inline docs/content/2.concepts/2.rendering.md:74]: Technically, this is a subset of server-side rendering in Workers and not serverless.

File: docs/content/2.concepts/2.rendering.md (hunk 3 above)
Comment (Atinux) [reply docs/content/2.concepts/2.rendering.md:17]: Well it is the same for Universal rendering, and we don't want to add to many pros for CSR

File: docs/content/2.concepts/2.rendering.md (hunk 7 above)
Comment (Atinux) [reply docs/content/2.concepts/2.rendering.md:74]: Fell free to push an update once this PR is merged 😄

File: docs/content/2.concepts/2.rendering.md (hunk 6 above)
Comment (Atinux) [reply docs/content/2.concepts/2.rendering.md:48]: Added :)

File: docs/content/2.concepts/2.rendering.md (hunk 3 above)
Comment (pi0) [reply docs/content/2.concepts/2.rendering.md:17]: But offline-first app support is not just easy with SSR rendering logic. Offline page handling alone is a nightmare. Let's be honest in docs than what we prefer ;)

Comment (Atinux) [reply docs/content/2.concepts/2.rendering.md:17]: You can add a line to it then, but offline is not supported by default in Nuxt 3 and shall be supported manually (for now?)

Comment (pi0) [reply docs/content/2.concepts/2.rendering.md]: Yeah. Phrased it as "it keeps working while is offline"

**Pull request merged**